manubot/pandoc_filter/cite.py

```diff
@@ -0,0 +1,102 @@
+"""
+Preliminary testing command:
+
+pandoc \
+  --filter=manubot/pandoc_filter/cite.py \
+  --to=markdown \
+  manubot/pandoc_filter/tests/input-with-cites.md
+
+pandoc \
+  --to=json \
+  manubot/pandoc_filter/tests/input-with-cites.md \
+  | python manubot/pandoc_filter/cite.py markdown
+"""
+
+"""
+https://github.com/jgm/pandocfilters
+https://github.com/sergiocorreia/panflute
+http://scorreia.com/software/panflute/code.html#panflute.elements.Citation
+"""
+import argparse
+import json
+import sys
+
+# import pandocfilters
+import panflute
+
+from manubot.cite.util import (
+    is_valid_citation_string,
+)
+from manubot.process.util import (
+    get_citation_df_from_strings,
+)
+
+
+global_variables = {
+    'citation_strings': list(),
+}
+
+
+def parse_args():
+    """
+    Read command line arguments
+    """
+    parser = argparse.ArgumentParser(description='Pandoc filter for citation by persistent identifier')
+    parser.add_argument('target_format')
+    parser.add_argument('--pandocversion', help='The pandoc version.')
```

Comment:
@tomduck, I copied this `--pandocversion` argument [from](https://github.com/tomduck/pandoc-fignos/blob/44776d885b8101d9d6aeb6845b17ad1625e88e6f/pandoc_fignos.py#L63) `pandoc-fignos`. Do you know what it's for and whether it's needed? It doesn't seem to me that pandoc 2.5 supplies this argument to filters.